app/file.go

```diff
@@ -310,21 +314,22 @@ func (a *App) DoUploadFile(now time.Time, rawTeamId string, rawChannelId string,
 	info.Path = pathPrefix + filename
 
 	if info.IsImage() {
```

Comment:
I'm thinking something like this:
```
	if info.IsImage() {
		nameWithoutExtension := filename[:strings.LastIndex(filename, ".")]
		// Check dimensions before loading the whole thing into memory later on
		if info.Size > MaxImageSize {
			err := model.NewAppError("uploadFile", "api.file.upload_file.large_image.app_error", map[string]interface{}{"Filename": filename}, "", http.StatusBadRequest)
			return nil, err
		} else if info.Width*info.Height > MaxImageSize {
			info.ThumbnailPath = GENERIC_THUMBNAIL_PATH
		} else {
			info.ThumbnailPath = pathPrefix + nameWithoutExtension + "_thumb.jpg"
		}

		info.PreviewPath = pathPrefix + nameWithoutExtension + "_preview.jpg"
	}
```

Reply:
Hi @saturninoabril!  Thanks for the review.  If the path stored for the thumbnail image is: `client/images/generic_thumbnail_image.png`.  Does this mean that any request for: `info.ThumbnailPath = GENERIC_THUMBNAIL_PATH` would not be made to storage, but instead to the app server's `client` directory?  I thought that by default, requests to info.ThumbnailPath` were made to storage, this is why I added the thumbnail to storage.  If you want to make the path on the app server side instead of storage, is there any necessary logic to be added, or will this function as you hope?

Reply:
Also, @saturninoabril, the first `if` conditional you added: `info.Size > MaxImageSize`, I believe this check already occurs here in [`api4/file.go`](https://github.com/mattermost/mattermost-server/blob/10c5a927cb619f1aa2a599cbe3667820f4766bda/api4/file.go#L67).  Did you still want to run this check a second time?

Reply:
1) Path of `generic_thumbnail_image.png`, I don't think it should reside to local or S3 storage. If file's thumbnail is equal to `GENERIC_THUMBNAIL_PATH` then it should not hit the `files/{file_id:[A-Za-z0-9]+}/thumbnail` api. The webapp (or client) should immediately render `generic_thumbnail_image.png`.
But @hmhealey might have a better approach to this.
2) You're correct. I missed that.